docs/theory.rst

```diff
@@ -16,79 +22,188 @@ The filter shape can be thought of in terms of the kernel of a convolution filte
 The default is ``transition_width`` = :math:`\pi`.
-Larger values reduce the cost and the likelihood of producing negative values from positive data, but make the filter less scale-selective.
+Larger values for ``transition_width`` reduce the cost and the likelihood of producing negative values from positive data, but make the filter less scale-selective. In the example below, we choose ``transition_width`` = :math:`2\pi`.
+
+.. ipython:: python
+
+    wider_taper_filter = gcm_filters.Filter(
+        filter_scale=4,
+        dx_min=1,
+        filter_shape=gcm_filters.FilterShape.TAPER,
+        transition_width=2*np.pi,
+        grid_type=gcm_filters.GridType.REGULAR,
+    )
+    wider_taper_filter
+
+.. ipython:: python
+    :okwarning:
+
+    @savefig wider_taper_shape.png
+    wider_taper_filter.plot_shape()
+
 
 Filter Steps
 ------------
 
 The filter goes through several steps to produce the final filtered field.
-There are two different kinds of steps: "Laplacian" and "Biharmonic" steps.
+There are two different kinds of steps: *Laplacian* and *Biharmonic* steps.
 At each Laplacian step, the filtered field is updated using the following formula
 
 .. math:: \bar{f} \leftarrow \bar{f} + \frac{1}{s_{j}}\Delta \bar{f}
 
 The filtered field is initialized to :math:`\bar{f}=f` and :math:`\Delta` denotes a discrete Laplacian.
 At each Biharmonic step, the filtered field is updated using
 
-.. math:: \bar{f}\leftarrow \bar{f}+\frac{2R\{s_j\}}{|s_j|^2}\Delta\bar{f} + \frac{1}{|s_j|^2}\Delta^2\bar{f}
+.. math:: \bar{f}\leftarrow \bar{f}+\frac{2R\{s_j\}}{|s_j|^2} \Delta\bar{f} + \frac{1}{|s_j|^2}\Delta^2\bar{f}
 
 where :math:`R\{\cdot\}` denotes the real part of a complex number.
 
-The total number of steps and the values of :math:`s_j` are automatically selected by the code to produce the desired filter scale and shape.
+The total number of steps, ``n_steps``, and the values of :math:`s_j` are automatically selected by the code to produce the desired filter scale and shape.
 If the filter scale is much larger than the grid scale, many steps are required.
-Also, the Taper filter requires more steps than the Gaussian filter for the same ``filter_scale``.
+Also, the Taper filter requires more steps than the Gaussian filter for the same ``filter_scale``; in the above examples the Taper filters required ``n_steps`` = 16, but the Gaussian filter only ``n_steps`` = 5.
 
-The code allows users to set their own number of steps ``n_steps``.
-The minimum number of steps is 3; if ``n_steps`` is not set by the user, or if it is set to a value less than 3, the code automatically changes ``n_steps`` to the default value.
-If the number of steps is set too low the filter will not behave as expected: it may not have the right shape or the right length scale.
-
-Biharmonic steps are counted as 2 steps because their cost is approximately twice as much as a laplacian step.
+The code allows users to set their own ``n_steps``.
```

Comment:
Somewhere in here we should say that the default number of steps computed by the code for the Taper filter assumes the default transition width. If the user selects a wider transition, then they might want to set a value of `n_steps` lower than the default to save cost while keeping accuracy. If the user selects a narrower transition then they should set a value of `n_steps` higher than the default, otherwise the filter won't behave as expected.

Reply:
Would it be straightforward to change the code to the effect that it takes the input transition width into account when computing the default `n_steps`?

Reply:
I set the scaling empirically, so I guess I could look for some kind of empirical scaling law that takes the transition width into account. I will put it on my to do list for the next few days.